modules/core/src/main/scala/doobie/util/fragments.scala

```diff
@@ -45,6 +47,45 @@ object fragments {
     NonEmptyList.fromFoldable(fs).map(nel => notIn(f, nel))
   }
 
+  @inline
+  private def inNotInValues[F[_]: Reducible: Functor, A](
+      fwop: Fragment,
+      fs: F[A],
+      fallback: Fragment
+  )(implicit A: util.Write[A]): Fragment = {
+    if (A.length == 0) fallback
+    else {
+      fwop ++ {
+        if (A.length == 1) // no need for extra parentheses
+          parentheses(comma(fs.map(values(_))))
+        else
+          parentheses(comma(fs.map(a => parentheses0(values(a)))))
+      }
+    }
+  }
+
+  /** Returns `f IN (fs0, fs1, ...)`.
+    * @param f
+    *   left-hand expression.
+    * @param fs
+    *   values of `Product` type to compare to the left-hand expression.
+    * @return
+    *   the `IN` subquery expression or `FALSE` if `fs` is a 0-arity product.
+    */
+  def inValues[F[_]: Reducible: Functor, A: util.Write](f: Fragment, fs: F[A]): Fragment =
```

Comment:
I'm also wondering whether we can replace `in` with this. It should be source compatible

Reply:
Do you mean, remove the existing `in`/`notIn` whatsoever and rename the proposed `inValues`/`notInValues` to `in`/`notIn`, correct?

Reply:
Yes that's right